### PR TITLE
Add parser for Snitch messages

### PR DESCRIPTION
**Pull request by Allda**
Snitch is a new service which emits Mongo database events to UMB.
So far there is only one type of messages.

JIRA: PYXIS-239

### CODE REVIEW

**Comment by Allda** [comment]
@mikebonnet Please take a look. Thanks

File: fedmsg_meta_umb/snitch.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    def subtitle(self, msg, **config):
+        inner_msg = msg["msg"]
+        headers = msg["headers"]
+
+        if not isinstance(inner_msg, dict):
+            return "Unknown message format"
+
+        template = self._(
+            "New DB event ({operationType}): {entityName} - {entityData}.{_id}.{$oid}"
```
Comment (mikebonnet) [inline fedmsg_meta_umb/snitch.py]: I don't think this format string will work for retrieving nested data. You want something like `{entityData[_id][$oid]}`.

**Comment by mikebonnet** [comment]
Aha! Forgot to add the `entry_point` in `setup.py`:

`snitch=fedmsg_meta_umb.snitch:SnitchProcessor`

**Comment by mikebonnet** [comment]
With the `entry_point` added, the processor is being matched to the test message, but I'm now getting this error:

`TypeError: format() got multiple values for keyword argument 'entityName'`

**Comment by Allda** [comment]
Lol and I was surprised that test passed with first execution :D I modified setup.py and processor. Now test really pass.

File: fedmsg_meta_umb/tests/test_snitch.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+class TestSnitch(fedmsg.tests.test_meta.Base):
+    """
+    Simple parser of Snitch messages
+    """
+
+    expected_title = "snitch"
+    expected_subti = "New DB event (replace): containerImage - 5e17f347dd19c77896f4f7c0"
+    expected_icon = "_static/img/icons/snitch.png"
+
+    {
```
[the remark this answers is not in the document]
Comment (Allda) [reply fedmsg_meta_umb/tests/test_snitch.py]: fixed

File: fedmsg_meta_umb/snitch.py (hunk 1 above)
Comment (Allda) [reply fedmsg_meta_umb/snitch.py]: fixed

File: fedmsg_meta_umb/snitch.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    __icon__ = "_static/img/icons/snitch.png"
+    __obj__ = "Snitch"
+
+    def title(self, msg, **config):
+        return msg["topic"].split(".", 2)[-1]
+
+    def subtitle(self, msg, **config):
+        inner_msg = msg["msg"]
+        headers = msg["headers"]
+        inner_msg["operationType"] = headers["operationType"]
```
Comment (mikebonnet) [inline fedmsg_meta_umb/snitch.py]: Could you make a copy of `inner_msg` before modifying it?

Comment (Allda) [reply fedmsg_meta_umb/snitch.py]: Done

**Pull request merged**